mantidproject/mantid
# 37107 findsxpeaks peak size validation

### Description of work
`FindSXPeaks` returns spurious peaks due to noise and peaks with a large number of contributing pixels due to powder lines. A set of new parameters were introduced to the algorithm to validate peaks as below.
- `MinBinsForAPeak`, the Minimum number of bins contributing to a peak in an individual spectrum
- `MinSpectrumsForAPeak`, `MaxSpectrumsForAPeak` Minimum & Maximum number of spectra contributing to a peak after they are grouped

Fixes #37107. 

### To test:

1. Load a workspace (ex: SXD23767)
2. Select FindSXPeaks algorithm from the algorithm pane
3. Select the above workspase as InputWorkspace
4. Pick a `PeakFindingStratergy` of interest, ex:-`AllPeaksNSigma` 
5. Provide values for the new parameters `MinBinsForAPeak`, `MinSpectrumsForAPeak`, and `MaxSpectrumsForAPeak` values for which are independent (other than the validation `MinSpectrumsForAPeak` < `MaxSpectrumsForAPeak`)
6. Enter a name for `OutputWorkspace`
7. Check the results in OutputWorkspace when those parameters are varied, the number of peaks detected should be filtered accordingly. 







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
